apps/chat_server/priv/repo/migrations/20171003225928_add_starred_messages_table.exs

```diff
@@ -0,0 +1,15 @@
+defmodule ChatServer.Repo.Migrations.AddStarredMessagesTable do
+  use Ecto.Migration
+
+  def change do
+    create table(:starred_messages, primary_key: false) do
+      add :user_id, references(:users, type: :uuid)
+      add :message_id, references(:messages, type: :uuid)
+      timestamps()
+    end
+
+    create index(:starred_messages, [:message_id])
+    create index(:starred_messages, [:user_id])
+    create unique_index(:starred_messages, [:user_id, :message_id], name: :starred_messages_user_message_index)
```

Comment:
Not a blocker, but one thing to be aware of with compound indexes like these is that often times the order is meaningful.

If a query is made asking for `user_id` then `message_id` then it'll use the index and do a fast lookup. If a query reverses the order and passes in `message_id` then `user_id` the index won't be used and a slower lookup will occur.

I'm not a database expert, so it's possible either Ecto optimizes to prevent an unoptimized query from happening, or that compound key ordering isn't an issue in PostgreSQL like it was in MySQL.